llama-index-integrations/tools/llama-index-tools-agentql/examples/AgentQL_browser_agent.ipynb

```diff
@@ -0,0 +1,817 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "os.environ[\"AGENTQL_API_KEY\"] = \"YOUR_AGENTQL_API_KEY\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Set up Playwright browser and AgentQL tools\n",
+    "To run this notebook, install Playwright browser and configure Jupyter Notebook's `asyncio` loop."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!playwright install\n",
+    "\n",
+    "# This import is required only for jupyter notebooks, since they have their own eventloop\n",
+    "import nest_asyncio\n",
+    "\n",
+    "nest_asyncio.apply()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Instantiation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### `AgentQLRestAPIToolSpec`\n",
+    "`AgentQLRestAPIToolSpec` provides `extract_web_data_with_rest_api` function tool.\n",
+    "\n",
+    "You can instantiate `AgentQLRestAPIToolSpec` with the following param:\n",
+    "- `timeout`: The number of seconds to wait for a request before timing out. Increase if data extraction times out. **Defaults to `900`.**\n",
+    "- `is_stealth_mode_enabled`: Whether to enable experimental anti-bot evasion strategies. This feature may not work for all websites at all times. Data extraction may take longer to complete with this mode enabled. **Defaults to `False`.**\n",
+    "- `wait_for`: The number of seconds to wait for the page to load before extracting data. **Defaults to `0`.**\n",
+    "- `is_scroll_to_bottom_enabled`: Whether to scroll to bottom of the page before extracting data. **Defaults to `False`.**\n",
+    "- `mode`: `\"standard\"` uses deep data analysis, while `\"fast\"` trades some depth of analysis for speed and is adequate for most usecases. [Learn more about the modes in this guide.](https://docs.agentql.com/accuracy/standard-mode) **Defaults to `\"fast\"`.**\n",
+    "- `is_screenshot_enabled`: Whether to take a screenshot before extracting data. Returned in 'metadata' as a Base64 string. **Defaults to `False`.**\n",
+    "\n",
+    "`AgentQLRestAPIToolSpec` is using AgentQL REST API, for more details about the parameters read [API Reference docs](https://docs.agentql.com/rest-api/api-reference)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_index.tools.agentql import AgentQLRestAPIToolSpec\n",
+    "\n",
+    "agentql_rest_api_tool = AgentQLRestAPIToolSpec()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### `AgentQLBrowserToolSpec`\n",
+    "`AgentQLBrowserToolSpec` provides 2 tools: `extract_web_data_from_browser` and `get_web_element_from_browser` function tools."
```

Comment:
you don't need to repeat "function tools" in the end, if you mentioned tools in the beginning